src/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -112,7 +62,7 @@ public int GetHashCode(Uri obj)
     /// the URI.
     /// <para/> Access to this is guaranteed to be serial by the <see cref="RequestExecutionQueue{RequestContextType}"/>
     /// </summary>
-    private ImmutableDictionary<Uri, (SourceText Text, string LanguageId)> _trackedDocuments = ImmutableDictionary<Uri, (SourceText, string)>.Empty.WithComparers(LspUriComparer.Instance);
+    private ImmutableDictionary<DocumentUri, (SourceText Text, string LanguageId)> _trackedDocuments = ImmutableDictionary<DocumentUri, (SourceText SourceText, string LanguageId)>.Empty;
```

Comment:
rest of this file is interesting - contains changes to attempt to parse and handle issues